server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -205,6 +205,7 @@ public Object startJob()
   public int add(InputRow row, Supplier<Committer> committerSupplier) throws IndexSizeExceededException
   {
     final Sink sink = getSink(row.getTimestampFromEpoch());
+    metrics.reportMessageMaxTimestamp(rejectionPolicy.getCurrMaxTime().getMillis());
```

Comment:
why timestamp is not taken from the row instead. above is only meaningful for message-time rejection policy in this case.